form.php

```diff
@@ -0,0 +1,86 @@
+<?php
+require_once "functions.php";
+
+$message = '';
+if (empty($_GET)) {
```

Comment:
Put all this ($message and if/else statement) into a function. Call it here and pass in `$_GET`; you won't have to unit test it.